XPNet.CLR/Scenery/XPlaneScenery.cs

```diff
@@ -84,7 +84,11 @@ public IXPInstance CreateInstance(IEnumerable<string> inDataRefs)
 
 		public void Draw(int lighting, int earthRelative, XPLMDrawInfo_t[] drawInfos)
 		{
-			PluginBridge.ApiFunctions.XPLMDrawObjects(m_objectRef, drawInfos.Length, drawInfos[0], lighting, earthRelative);
+			fixed (void* p = &drawInfos[0].m_structSize)
```

Comment:
@jaurenq I am trying to figure out how to pass a reference to the array of XPLMDrawInfo_t (which I will rename later on according to your request) and somehow I am not getting it right yet. Can I pin it like this, or can't I? When I am running the XPNetPlugInTestHost, only the positions of the first instance are correct. The rest is supposedly uninitialized data.

Reply:
You need to make XPLMDrawInfo_t be a struct, so that .NET will lay out arrays of XPLMDrawInfo_t objects contiguously in memory the way C would.  As a class, instances of XPLMDrawInfo_t are allocated on the heap (meaning, in essentially random memory locations) and an array of them is actually an array of pointers to their real locations, which is not what the X-Plane side is expecting.

I am not where I can run it this morning but I'd expect it to work with the following changes:

1. Change XPLMDrawInfo_t to a struct: `public struct XPLMDrawInfo_t`

2. Change the declaration of XPLMDrawObjects to accept a pointer to XPLMDrawInfo_t.  (void* would work, but we loose some type safety and AFAICT we don't gain anything for it).

```
	[UnmanagedFunctionPointer(CallingConvention.Cdecl)]
	internal unsafe delegate void XPLMDrawObjects(
		void* inObject,
		int inCount,
		XPLMDrawInfo_t* inLocations,
		int lighting,
		int earth_relative
	);
```

3. Now that XPLMDrawInfo_t is a struct, you'll be able to pin the entire array, which is what you want.

```
			fixed (XPLMDrawInfo_t* p = &drawInfos[0])
			{
				PluginBridge.Log.Log($"Drawing {drawInfos.Length} objects");
				PluginBridge.ApiFunctions.XPLMDrawObjects(m_objectRef, drawInfos.Length, p, lighting, earthRelative);
			}
```

Reply:
And actually I _think_ the 'fixed' line in #3 above could be written like this as well: 

```
fixed (XPLMDrawInfo_t* p = drawInfos) { ... }
```

But either is fine and anyone who knows C will know what's going on w/ the `&drawInfos[0]` version.

Reply:
@jaurenq you are great, that works like a breeze. I was not aware that structs and classes were handled that different internally...